### PR TITLE
Show connection value instead of json on task node

### PR DESCRIPTION
**Pull request by Mbeaulne**
## Description
closes: https://github.com/Cloud-Pipelines/pipeline-studio-app/issues/294
This pull request enhances how input values are displayed in task node handles, making it clearer when a value is connected to another node versus being a literal value. It introduces a new utility function for formatting these values, updates the UI to leverage tooltips for better readability, and provides thorough test coverage for the new logic.

**Display improvements for input handle values:**

* Added a new `getDisplayValue` utility in `handleUtils.ts` to format input values, showing connected nodes as `→ ComponentName.outputName` and formatting other values for readability. (`src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/handleUtils.ts`, [src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/handleUtils.tsR1-R34](diffhunk://#diff-242c91a996a2e13b08e80aa1b507875e784bb048056542ec593714cc2bdb8144R1-R34))
* Updated `TaskNodeInputs.tsx` to use `getDisplayValue` instead of the previous `getValue` function, ensuring all input values are displayed using the new logic. (`src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/TaskNodeInputs.tsx`, [[1]](diffhunk://#diff-28779bf173416906aa25e2ebd88bb290c81c6f6659af3fbe2f3b146dd3d70b1aR7-R16) [[2]](diffhunk://#diff-28779bf173416906aa25e2ebd88bb290c81c6f6659af3fbe2f3b146dd3d70b1aR30) [[3]](diffhunk://#diff-28779bf173416906aa25e2ebd88bb290c81c6f6659af3fbe2f3b146dd3d70b1aL221-R223)

**UI/UX enhancements:**

* Wrapped input values in the handle UI with a tooltip, so users can see the full value on hover, improving usability for truncated or complex values. (`src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/Handles.tsx`, [[1]](diffhunk://#diff-ea82af7ae9ac6ca19a7f4b18bb5503343fd68f6a25e9d2dc6857abbd06443ab3R9-R13) [[2]](diffhunk://#diff-ea82af7ae9ac6ca19a7f4b18bb5503343fd68f6a25e9d2dc6857abbd06443ab3R172-R173) [[3]](diffhunk://#diff-ea82af7ae9ac6ca19a7f4b18bb5503343fd68f6a25e9d2dc6857abbd06443ab3R182-R188)

**Testing and reliability:**

* Added comprehensive unit tests for `getDisplayValue`, covering connected node values, literals, edge cases, and fallback behaviors. (`src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/handleUtils.test.ts`, [src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/handleUtils.test.tsR1-R221](diffhunk://#diff-44d6054e7c5444f65cadd6a7ce006d56c5f87022594d45711259996abebfe910R1-R221))

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Improvement
- [ ] Cleanup/Refactor
- [ ] Breaking change
- [ ] Documentation update

## Checklist

- [ ] I have tested this does not break current pipelines / runs functionality
- [ ] I have tested the changes on staging

## Test Instructions
- Create a pipeline or view run
- Confirm inputs that are connected to other cards show the new design.

## Additional Comments

Screenshot 2025-08-21 at 5 11 11 PM
Screenshot 2025-08-21 at 5 11 22 PM

### CODE REVIEW

File: src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/Handles.tsx [hunk 1]
```diff
@@ -164,14 +169,23 @@ export const InputHandle = ({
...
             >
-              <div
-                className={cn(
-                  "text-xs text-gray-800! truncate inline-block text-right pr-2",
-                  !hasValue && "text-gray-400! italic",
-                )}
-              >
-                {hasValue ? value : input.default}
-              </div>
+              <Tooltip>
```
Comment (Mbeaulne) [inline src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/Handles.tsx:172]: I know we discussed tool tips, but until we have a solution I am going to use our current patterns.

File: src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/Handles.tsx [hunk 2]
```diff
@@ -164,14 +169,23 @@ export const InputHandle = ({
...
+                <TooltipTrigger asChild>
+                  <div
+                    className={cn(
+                      "text-xs text-gray-800! truncate inline-block text-right pr-2",
+                      !hasValue && "text-gray-400! italic",
+                    )}
+                  >
+                    {hasValue ? value : input.default}
```
Comment (camielvs) [inline src/components/shared/ReactFlow/FlowCanvas/TaskNode/TaskNodeCard/Handles.tsx:180]: this is done twice. good enough reason to make a const out of it

**Review by camielvs — APPROVED** [review]
I think we may be able to style a little better than a vanilla arrow → to represent this at a glance, but for now the arrow is much, much better than what we currently have so let's ship this
